Mixpanel/Persistence.swift

```diff
@@ -77,14 +77,19 @@ class Persistence {
     #endif // DECIDE
 
     class func archiveEvents(_ eventsQueue: Queue, token: String) {
+        objc_sync_enter(self)
```

Comment:
if we are using a lock now, do we still need these to use a different lock on self? do we not want them synced with the other stuff?